Block Editor: Renaming Bugfixes

chynu opened the pull request:
When renaming a block in the block editor to an already taken block name, devtools previously threw an error which broke the application (couldn't switch between blocks). This bug is now fixed, and alerts the user if they try to name a block with an invalid block name.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly-devtools/209)
<!-- Reviewable:end -->

chynu committed:
Debugging rename

chynu committed:
Attempting to fix rename issue

chynu committed:
Error handling for renaming to duplicates

chynu committed:
Nits

chynu committed:
Merge branch 'develop' into naming

chynu committed:
warning

chynu committed:
bugfixes

chynu committed:
Nits

AnmAtAnm reviewed (approved):
Minor suggestions.
Otherwise LGTM

AnmAtAnm commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/controller/block_editor_controller.js, line 402 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/209#-Kr2eVRidp5S-BWDi67g:-Kr2eVRidp5S-BWDi67h:b-sdwxfe) ([raw file](https://github.com/google/blockly-devtools/blob/cab4705ef80dfb697bbea1cc2802607baca18836/src/controller/block_editor_controller.js#L402)):*
> ```JavaScript
>       return 'There is already a block under this name.';
>     } else {
>       return '';
> ```

Is this a non-warning? Is this an unrecognized warnable state worth logging?  Comment or log appropriately.

---

*[src/controller/new_block_popup_controller.js, line 94 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/209#-Kr2eepmga0Uql3dfIIE:-Kr2eepmga0Uql3dfIIF:b9mp43t) ([raw file](https://github.com/google/blockly-devtools/blob/cab4705ef80dfb697bbea1cc2802607baca18836/src/controller/new_block_popup_controller.js#L94)):*
> ```JavaScript
>       this.view.showWarning(isDuplicate, `This block name is already taken in
>           this project.`);
>       this.view.setEnable(false);
> ```

Rename to `setEnabled(..)`. Adjective form instead of verb form.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/209)*
<!-- Sent from Reviewable.io -->

chynu committed:
PR responses + bugfix

chynu commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/controller/block_editor_controller.js, line 402 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/209#-Kr2eVRidp5S-BWDi67g:-Kr2fqVegl7KJh9WEG0C:bcsrgg3) ([raw file](https://github.com/google/blockly-devtools/blob/cab4705ef80dfb697bbea1cc2802607baca18836/src/controller/block_editor_controller.js#L402)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Is this a non-warning? Is this an unrecognized warnable state worth logging?  Comment or log appropriately.
</blockquote></details>

Updated jsdoc, and changed it to return `null`.

---

*[src/controller/new_block_popup_controller.js, line 94 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/209#-Kr2eepmga0Uql3dfIIE:-Kr2k3hKeXLBIwSBCVQT:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/cab4705ef80dfb697bbea1cc2802607baca18836/src/controller/new_block_popup_controller.js#L94)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Rename to `setEnabled(..)`. Adjective form instead of verb form.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/209)*
<!-- Sent from Reviewable.io -->

chynu committed:
bugfix

chynu committed:
Merge branch 'develop' into renaming
